packages/atlas-service/src/main.ts

```diff
@@ -524,6 +533,64 @@ export class AtlasService {
     await throwIfNotOk(res);
   }
 
+  static async getAIFeatureEnablement(): Promise<AIFeatureEnablement> {
+    throwIfNetworkTrafficDisabled();
+
+    const userId = (await this.getActiveCompassUser()).id;
+
+    const res = await this.fetch(
+      `${this.config.atlasApiBaseUrl}/ai/api/v1/hello/${userId}`
+    );
+
+    await throwIfNotOk(res);
+
+    const body = await res.json();
+
+    validateAIFeatureEnablementResponse(body);
+
+    return body;
+  }
+
+  static async setupAIAccess(): Promise<void> {
+    log.info(
+      mongoLogId(1_001_000_227),
+      'AtlasService',
+      'Fetching if the AI feature is enabled'
+    );
+
+    try {
+      throwIfNetworkTrafficDisabled();
+
+      const featureResponse = await this.getAIFeatureEnablement();
+
+      const isAIFeatureEnabled =
+        !!featureResponse.features.GEN_AI_COMPASS?.enabled;
+
+      log.info(
+        mongoLogId(1_001_000_229),
+        'AtlasService',
+        'Fetched if the AI feature is enabled',
+        {
+          enabled: isAIFeatureEnabled,
+        }
+      );
+
+      await preferences.savePreferences({
+        cloudFeatureRolloutAccess: {
+          GEN_AI_COMPASS: isAIFeatureEnabled,
+        },
+      });
+    } catch (err) {
+      // Default to what's already in Compass when we can't fetch the preference.
+      log.error(
+        mongoLogId(1_001_000_243),
+        'AtlasService',
+        'Failed to load if the AI feature is enabled',
+        { error: (err as Error).stack }
+      );
+    }
+  }
+
   static async getAggregationFromUserInput({
```

Comment:
These changes should be accounted for in `throwIfAINotEnabled` that is used inside AI endpoint methods so that atlas service won't allow to call these endpoints

Reply:
Good call!